app/scripts/modules/core/src/managed/StatusBubble.tsx

```diff
@@ -36,16 +38,47 @@ const transitionConfig = {
   config: { mass: 1, tension: 400, friction: 30 },
 };
 
-export const StatusBubble = memo(({ iconName, appearance, size }: IStatusBubbleProps) => {
+export const StatusBubble = memo(({ appearance, iconName, size, quantity }: IStatusBubbleProps) => {
   const transitions = useTransition(iconName, null, transitionConfig);
+  let quantityPill: ReactNode = null;
+  let withQuantity = false;
+
+  // Render quantity text only for size >= small.
+  if (quantity && (size === 'small' || size === 'medium' || size === 'large' || size === 'extraLarge')) {
+    withQuantity = true;
+    quantityPill = (
+      <span
+        className={classnames({
+          Pill: true,
+          [size]: true,
+          [`status-${appearance}`]: true,
+        })}
+      >
+        {quantity}
+      </span>
+    );
+  }
 
   return (
-    <div className={`StatusBubble status-bubble-${appearance} sp-padding-${paddingBySize[size]}`}>
+    <>
       {transitions.map(({ item, key, props }) => (
-        <animated.div key={key} className="status-bubble-icon-container flex-container-h center middle" style={props}>
-          <Icon name={item} appearance="light" size={size} />
+        <animated.div className="StatusBubble" key={key} style={props}>
+          <div
+            className={classnames({
+              IconWrapper: true,
+              [`sp-padding-${paddingBySize[size]}`]: true,
+              [`status-${appearance}`]: true,
+              withQuantity,
```

Comment:
fwiw `classnames` takes variadic args which can eliminate a lot of the redundant `true`s in object notation when you're mixing static classes with dynamic — e.g.:
```tsx
className={classnames(
  'IconWrapper',
  `sp-padding-${paddingBySize[size]}`,
  `status-${appearance}`,
  { withQuantity }
})}
```